app.go

```diff
@@ -113,6 +113,20 @@ func Options(opts ...Option) Option {
 	})
 }
 
+// Printer is the interface required by fx's logging backend. It's implemented
+// by most loggers, including the standard library's.
```

Comment:
`(type *zap.Logger has no field or method Printf)`

If we wanted to use zap as an actual internal logger for the app as well, would we just build a bridge?